cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/TaskWorkerService.java

```diff
@@ -100,6 +103,7 @@ protected void startUp() throws Exception {
     preloadArtifacts();
 
     LOG.debug("Starting TaskWorkerService");
+    metricsCollectionService.startAndWait();
```

Comment:
Don't start a service that passed to you. It should be started at the main class level (or the class that creates the injector).

Reply:
Done